contrib/opentimelineio_contrib/adapters/advanced_authoring_format.py

```diff
@@ -986,6 +1148,79 @@ def _fix_transitions(thing):
             _fix_transitions(child)
 
 
+def _attach_markers(collection):
+    """Search for markers on tracks and attach them to their corresponding item.
+
+    Marked ranges will also be transformed into the new parent space.
+
+    """
+    # iterate all timeline objects
+    for timeline in collection.each_child(descended_from_type=otio.schema.Timeline):
+        tracks_map = {}
+
+        # build track mapping
+        for track in timeline.each_child(descended_from_type=otio.schema.Track):
+            metadata = track.metadata.get("AAF", {})
+            slot_id = metadata.get("SlotID")
+            track_number = metadata.get("PhysicalTrackNumber")
+            if slot_id is None or track_number is None:
+                continue
+
+            tracks_map[(int(slot_id), int(track_number))] = track
+
+        # iterate all tracks for their markers and attach them to the matching item
+        for current_track in timeline.each_child(descended_from_type=otio.schema.Track):
+
+            # get top level item
+            top_level = current_track
+            while top_level.parent() is not None:
+                top_level = top_level.parent()
+
+            for marker in list(current_track.markers):
+                metadata = marker.metadata.get("AAF", {})
+                slot_id = metadata.get("AttachedSlotID")
+                track_number = metadata.get("AttachedPhysicalTrackNumber")
+                target_track = tracks_map.get((slot_id, track_number))
+                if target_track is None:
+                    raise AAFAdapterError(
+                        "Marker '{}' cannot be attached to an item. SlotID: '{}', "
+                        "PhysicalTrackNumber: '{}'".format(
+                            marker.name, slot_id, track_number
+                        )
+                    )
+
+                # remove marker from current parent track
+                current_track.markers.remove(marker)
+
+                # determine new item to attach the marker to
+                target_item = target_track.child_at_time(marker.marked_range.start_time)
+                if target_item is None:
+                    target_item = target_track
+
+                # attach marker to target item
+                target_item.markers.append(marker)
+
+                # transform marked range into new item range
+                marked_start_global = current_track.transformed_time(
+                    marker.marked_range.start_time, top_level)
+                marked_start_local = top_level.transformed_time(marked_start_global,
+                                                                target_item)
```

Comment:
I think `transformed_time` could handle this in one step:
`marked_start_local = current_track.transformed_time(marker.marked_range.start_time, target_item)`